src/Microsoft.DotNet.ProjectJsonMigration/ProjectMigrator.cs

```diff
@@ -35,6 +35,8 @@ public void Migrate(MigrationSettings rootSettings, bool skipProjectReferences =
             Exception exc = null;
             IEnumerable<ProjectDependency> projectDependencies = null;
 
+            var tempMSBuildProjectTemplate = rootSettings.MSBuildProjectTemplate.DeepClone();
```

Comment:
Why do we `DeepClone()` here and call DeepClone() each time in the loop below?  Seems like this one could be removed.